pyproject.toml

```diff
@@ -52,9 +50,6 @@ pytest-describe = "^2"
 pyspark = ">3"
 semver = "^3"
 
-[tool.poetry.group.linting.dependencies]
-ruff = "^0.5"
```

Comment:
I'm using ruff, integrated into my IDE. And the ruff's binary is taken from the project venv. Why do we need to remove the version from this place?

Reply:
I think it is bad practice to have it both here and in the pre-commit hooks; this runs the risk of getting conflicting versions of ruff., which is also the case currently. I think there are three options:
- Don't use ruff in the IDE; the pre-commit hooks should take care of linting and formatting
- Let the pre-commit hooks use ruff from the virtual environment.
- Leave ruff in the dev dependencies and take care to keep the versions within pre-commit hooks and pyproject.toml in sync. 

I think option (1) and (3) are best, let me know which ones have your preference.

Reply:
The fact that we use different versions locally and in the CI/CD pipeline is now causing issues here; https://github.com/fpgmaas/quinn/actions/runs/9935477014/job/27441721396

Reply:
Sure, that sounds good! I updated the PR. We can wait for https://github.com/MrPowers/quinn/pull/244 and I can rebase afterwards.